consensus/scp/src/node.rs

```diff
@@ -9,31 +9,35 @@ use crate::{
 };
 use mc_common::{
     logger::{log, Logger},
-    LruCache, NodeID,
+    NodeID,
 };
 #[cfg(test)]
 use mockall::*;
-use std::{collections::BTreeSet, fmt::Display, time::Duration};
-
-/// Max number of pending slots to store.
-const MAX_PENDING_SLOTS: usize = 10;
+use std::{
+    collections::{BTreeSet, HashMap},
+    fmt::Display,
+    time::Duration,
+};
 
-/// Max number of externalized slots to store.
-const MAX_EXTERNALIZED_SLOTS: usize = 10;
+/// Default limit on number of externalized slots to store.
+const MAX_EXTERNALIZED_SLOTS: usize = 1;
```

Comment:
is there a disadvantage to keeping more slots around?

Reply:
I'm not sure yet, but I think the cost of each additional externalized slot is pretty low.

I have a hunch that we should rate-limit the production of new blocks (say, no more than one per second), then size the queue of externalized slots so that each node responds to messages for X blocks (i.e. X seconds) of recent data. That just makes it easy to reason about data availability, when to perform catch-up, etc.